crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1268,7 +1268,10 @@ impl World {
     /// Note that any resource with the [`Default`] trait automatically implements [`FromWorld`],
     /// and those default values will be here instead.
     #[inline]
+    #[track_caller]
     pub fn init_resource<R: Resource + FromWorld>(&mut self) -> ComponentId {
+        #[cfg(feature = "track_change_detection")]
+        let caller = core::panic::Location::caller();
```

Comment:
```suggestion
        let caller = Location::caller();
```

`Location` is imported, you don't need to qualified syntax :)